R/plot.R

```diff
@@ -68,57 +68,196 @@ fldts2df <- function(fldts, griddata)
+                   long = if_else(long > 180, long - 360, long)) ->
+            countries
+
+        ## TODO: make these options a little more customizable
+        if(nb < 2) {
+            ggplot2::ggplot(tdf, aes(x=lon, y = lat, fill = value)) +
+                ggplot2::geom_raster() +
+                ggplot2::scale_fill_distiller(palette=palettestr, direction=palettedir,
+                                              limits=c(minval, maxval), oob=scales::squish,
+                                              guide=ggplot2::guide_colorbar(title=legendstr, title.position='top')) +
+                ggplot2::scale_x_continuous(expand=c(0,0)) +
+                ggplot2::scale_y_continuous(expand=c(0,0)) +
+                ggplot2::theme(axis.title=element_blank(),
+                              axis.text=element_blank(),
+                              axis.ticks=element_blank(),
+                              legend.position = 'bottom') +
+                # black outlines of countries
+                ggplot2::geom_path(data = countries,
+                                   aes(x = long, y = lat, group = group),
+                                   size = 0.25, color = 'black')
+
+            # # Legacy code using gcammaptools
+            # gcammaptools::plot_GCAM_grid(tdf, col='value', extent=gcammaptools::EXTENT_WORLD,
+            #                              legend=TRUE) +
+            #     ggplot2::scale_fill_distiller(palette=palettestr, direction=palettedir,
+            #                                   limits=c(minval, maxval), oob=scales::squish,
+            #                                   guide=ggplot2::guide_colorbar(title=legendstr, title.position='top'))
+
+        }
+        else {
+            ## Discretize the output values.
+            tdf$value <- minval +
+              findInterval(tdf$value, seq(minval, maxval, length.out=nb))/nb * (maxval-minval)
+
+            ggplot2::ggplot(tdf, aes(x=lon, y = lat, fill = value)) +
+                ggplot2::geom_raster() +
+                ggplot2::scale_fill_distiller(palette=palettestr, direction=palettedir,
+                                              limits=c(minval, maxval), oob=scales::squish,
+                                              guide=ggplot2::guide_colorbar(title=legendstr, title.position='top')) +
+                ggplot2::scale_x_continuous(expand=c(0,0)) +
+                ggplot2::scale_y_continuous(expand=c(0,0)) +
+                ggplot2::theme(axis.title=element_blank(),
+                               axis.text=element_blank(),
+                               axis.ticks=element_blank(),
+                               legend.position = 'bottom') +
+                # black outlines of countries
+                ggplot2::geom_path(data = countries,
+                                   aes(x = long, y = lat, group = group),
+                                   size = 0.25, color = 'black')
+        }
+
```

Comment:
What is returned by this function? Although `return(plot)` isn't technically necessary CV has been encouraging me to explicitly return things in functions because it  is clearer and makes the R code easier for others with different backgrounds to read/understand the code.